source/acquaman/AMContinuousScanActionController.cpp

```diff
@@ -0,0 +1,513 @@
+			if(!foundScalerMovementEnd && encoderUpVector.at(x) > 20){
+				qDebug() << "Found scaler movement end at index " << x;
+				foundScalerMovementEnd = true;
+			}
+			else if(foundScalerMovementEnd && encoderUpVector.at(x) < 1){
+				foundScalerMovementStart = true;
+				scalerInitiateMovementIndex = x;
+				qDebug() << "Found scaler movement start index " << scalerInitiateMovementIndex;
+			}
+		}
+	}
+	else {
+		for(int x = encoderDownVector.count()-1; (x > 0) && !foundScalerMovementStart; x--){
+			if(!foundScalerMovementEnd && encoderDownVector.at(x) > 20){
+				qDebug() << "Found scaler movement end at index " << x;
+				foundScalerMovementEnd = true;
+			}
+			else if(foundScalerMovementEnd && encoderDownVector.at(x) < 1){
+				foundScalerMovementStart = true;
+				scalerInitiateMovementIndex = x;
+				qDebug() << "Found scaler movement start index " << scalerInitiateMovementIndex;
+			}
+		}
+	}
+
+	/* LEAVE THIS IN PLACE IN CASE WE HAVE TO GO BACK TO USING SCALER AS BASE TIME SCALE FOR EVERYTHING
+	for(int x = encoderUpVector.count()-6; (x > 5) && !foundScalerMovementStart; x--){
+
+		double runningAverage = (encoderUpVector.at(x+5)+encoderUpVector.at(x+4)+encoderUpVector.at(x+3)+encoderUpVector.at(x+2)+encoderUpVector.at(x+1)
+					 +encoderUpVector.at(x)+encoderUpVector.at(x-1)+encoderUpVector.at(x-2)+encoderUpVector.at(x-3)+encoderUpVector.at(x-4))/10;
+
+		if(!foundScalerMovementEnd && runningAverage > 1){
+			qDebug() << "Found scaler movement end at index " << x;
+			foundScalerMovementEnd = true;
+		}
+		else if(foundScalerMovementEnd && runningAverage < 0.1){
+			foundScalerMovementStart = true;
+			scalerInitiateMovementIndex = x;
+			qDebug() << "Found scaler movement start index " << scalerInitiateMovementIndex;
+		}
+	}
+	*/
+
+	qDebug() << "Amptek: " << initiateMovementIndex;
+	qDebug() << "Scaler: " << scalerInitiateMovementIndex;
+
+	if(!metaDataMap_.contains("GratingEncoderFeedback")){
+		qDebug() << "FATAL ERROR, MISSING READBACK VALUE FOR INITIAL ENCODER POSITION";
+		return;
+	}
+
+	int startEncoderValue = (int)(metaDataMap_.value("GratingEncoderFeedback"));
+	int currentEncoderValue = startEncoderValue;
+	QVector<double> scalerEnergyFeedbacks = QVector<double>(encoderUpVector.count()-scalerInitiateMovementIndex+1);
+	scalerEnergyFeedbacks[0] = SGMGratingSupport::energyFromGrating(SGMGratingSupport::LowGrating, startEncoderValue);
+
+	// Loop from the start of the movement to the end and recreate the axis values (energy in this case) from the encoder pulse changes
+	for(int x = scalerInitiateMovementIndex, size = encoderUpVector.count(); x < size; x++){
+		currentEncoderValue += encoderUpVector.at(x) - encoderDownVector.at(x);
+		scalerEnergyFeedbacks[x-scalerInitiateMovementIndex+1] = SGMGratingSupport::energyFromGrating(SGMGratingSupport::LowGrating, currentEncoderValue);
+	}
+
+	bool upScan = false;
+	if(scalerEnergyFeedbacks.at(0) < scalerEnergyFeedbacks.at(scalerEnergyFeedbacks.count()-1))
```

Comment:
Not important, but this could be `if (scalerEnergyFeedbacks.first() < scalerEnergyFeedbacks.last())`